agent/rpc/peering/service.go

```diff
@@ -181,13 +181,29 @@ func (s *Server) GenerateToken(
 		return nil, err
 	}
 
-	canRetry := true
-RETRY_ONCE:
-	id, err := s.getExistingOrCreateNewPeerID(req.PeerName, req.Partition)
+	peeringOrNil, err := s.getExistingPeering(req.PeerName, req.Partition)
 	if err != nil {
 		return nil, err
 	}
 
+	// validate that this peer name is not being used as a dialer already
+	if err = s.validatePeerRole(peeringOrNil, false); err != nil {
+		return nil, err
+	}
+
+	var id string
+	if peeringOrNil != nil {
+		id = peeringOrNil.ID
+	} else {
+		id, err = lib.GenerateUUID(s.Backend.CheckPeeringUUID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	canRetry := true
+RETRY_ONCE:
```

Comment:
The fetching and ID validation must be done under the RETRY_ONCE block

Reply:
Why? 🤔 

I think I see that if we don't do that then we could use the wrong ID right?

Just wanted to double check.

Reply:
If you read the comments below it intends for a retry to re-read from the state store and get the existing peer ID.

Otherwise you're retrying a failed write request once again with no changes in input

Reply:
Actually, maybe this is reason enough to keep the ID generation and validation seprate with their own calls to the state store 🤔 Wouldn't want to couple the validation with the retry logic

Reply:
Yeah we can use the old func as is then 🤔